### PR TITLE
Display WooCommerce styles during JP Onboarding when coming from the setup wizard

### PR DESCRIPTION
**Pull request by justinshreve**
This PR detects when we are in the Jetpack onboarding wizard, but have entered through the WooCommerce setup wizard. This is so we can do some co-branding, and keep the connection experience consistent throughout the store setup process. See p6riRB-4rG-p2 and https://github.com/woocommerce/woocommerce-admin/issues/2152.

This is step one of the process. It only adds the ability to target styles for this case, and implements the base background/button styles. It does not update the copy yet, or display the Woo logo, etc. It is also behind a feature flag so we can iterate on these differences without affecting the current flows.

@jameskoster I believe I could use some help here making sure I'm using the correct colors, hovers, accents, etc. I only had the background, text color, and button color to work with. See the mixin.

To Test:

* Run `npm run test-client` and make sure all tests pass.
* Disconnect Jetpack `Jetpack > Dashboard > Connections > Site connection > Manage site connection > Disconnect`.
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.
* Hit enter, and you should see Woo Purples
* Disconnect
* Logout
* Try the same same URL logged out
* Test the standard Jetpack connection process, without the WooCommerce query string appended, and verify you see the standard Jetpack Onboarding styles.

Screenshots:

Screen Shot 2019-05-20 at 2 56 57 PM
Screen Shot 2019-05-20 at 2 57 14 PM

### CODE REVIEW

File: client/jetpack-connect/signup.js [hunk 1]
```diff
@@ -85,10 +85,16 @@ export class JetpackSignup extends Component {
 		} );
 	}
 
+	isWoo( props = this.props ) {
```
Comment (tyxla) [inline client/jetpack-connect/signup.js]: Could we just use `const { authQuery } = this.props;` inside the method instead? It doesn't seem like we're using this as an argument anywhere.

File: client/jetpack-connect/signup.js [hunk 2]
```diff
@@ -85,10 +85,16 @@ export class JetpackSignup extends Component {
...
+		const { from } = props.authQuery;
```
Comment (tyxla) [inline client/jetpack-connect/signup.js]: Should we alias `from` here and everywhere else where it occurs, as it's a reserved keyword?

File: client/jetpack-connect/signup.js (hunk 1 above)
Comment (justinshreve) [reply client/jetpack-connect/signup.js]: Handled this and the other piece of feedback in https://github.com/Automattic/wp-calypso/pull/32993/commits/260bfd5c573b6916d8db91ba24ca33bea66dd254.

**Comment by justinshreve** [comment]
I've been making sure to run through this PR with the normal flow as well (no `&from=` override) as well as testing this with the feature flag disabled. All looks well.

I've adjusted the testing directions to make sure we get some more 👀 on that as well.

**Comment by justinshreve** [comment]
I've addressed the color feedback over in the other PR.

**Pull request merged**